src/NLog/Common/InternalLogger.cs

```diff
@@ -284,6 +291,36 @@ private static void Write([CanBeNull]Exception ex, LogLevel level, string messag
             }
         }
 
+        private static bool IsSeriousException(Exception ex)
+        {
+            return ex != null && ex.MustBeRethrownImmediately();
+        }
+
+        private static bool NoLogger()
+        {
+            return string.IsNullOrEmpty(LogFile) && !LogToConsole && !LogToConsoleError && LogWriter == null && !WriteToDiagnostics;
+        }
+
+        private static void LogToDiagnostics(LogLevel level, string msg)
+        {
+#if !SILVERLIGHT && !__IOS__ && !__ANDROID__
+            // log to System.Diagnostics.Debug / System.Diagnostics.Trace
+            if (!WriteToDiagnostics)
+            {
+                return;
+            }
+
+            System.Diagnostics.Debug.WriteLine(msg);
+
+            if (level != LogLevel.Trace)
+            {
+                return;
+            }
+
+            System.Diagnostics.Trace.WriteLine(msg);
```

Comment:
This is a bug? We will write to both Debug and Trace. 

Also Loglevel.Off seems to write to trace. (although I can be checked somewhere else)